### PR TITLE
[Vue] migrate report deletion controllers/store to Vue

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Changes:
* Migrate report-deletion-settings service to Vue store.
* Migrate delete-old-reports controller to Vue.
* Migrate delete-old-logs controller to Vue.
* Migrate scheduled-report-deletion controller to Vue.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/PrivacyManager/vue/src/ReportDeletionSettings/ReportDeletionSettings.store.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    this.privateState.isModified = false;
+    return AjaxHelper.post(
+      {
+        module: 'API',
+        method: apiMethod,
+      },
+      {
+        ...settings,
+        enableDeleteLogs: settings.enableDeleteLogs ? '1' : '0',
+        enableDeleteReports: settings.enableDeleteReports ? '1' : '0',
```
Comment (sgiehl) [inline plugins/PrivacyManager/vue/src/ReportDeletionSettings/ReportDeletionSettings.store.ts:79]: This doesn't seem to work properly. For me the request to calculate the stats always only include `enableDeleteLogs` and  `enableDeleteReports`, all other settings are missing and so the calculated value isn't correct.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left a comment. Everything else seems to work correctly.

File: plugins/PrivacyManager/vue/src/ReportDeletionSettings/ReportDeletionSettings.store.ts (hunk 1 above)
Comment (diosmosis) [reply plugins/PrivacyManager/vue/src/ReportDeletionSettings/ReportDeletionSettings.store.ts:79]: @sgiehl thanks for catching this :+1:, should be fixed. Also tested the manual estimate feature.

**Review by sgiehl — APPROVED** [review]
Seems to work as expected now. I have merged in latest 4.x-dev changes. Once tests are passing this is good to merge.

**Pull request merged**